File: MyungJi/240101_week/programmers_150369_택배배달과수거하기.cpp
// 내 코드 아님
#include <bits/stdc++.h>

using namespace std;

long long solution(int cap, int n, vector<int> deliveries, vector<int> pickups) {
    long long answer = 0;

    int del = 0;
    int pic = 0;
    for (int i = n - 1; i >= 0; i--)
    {
        if (deliveries[i] != 0 || pickups[i] != 0)
        {
            int cnt = 0;
            while (del < deliveries[i] || pic < pickups[i])
            {
                ++cnt;
                del += cap;
                pic += cap;
            }
            del -= deliveries[i];
            pic -= pickups[i];
            answer += (1LL * cnt) * (1LL * (i + 1)) * 2;
        }
    }

    return answer;
}

File: MinJu/240101_week/Programmers_150369_택배 배달과 수거하기.py
# 택배 배달과 수거하기

# 트럭에 재활용 상자 최대 cap개 올릴 수 있음
# 각 집에 배달/수거 최소 거리로 이동

# 가장 먼 곳부터 이동하기

def solution(cap, n, deliveries, pickups):
    answer = 0 # 최소 거리
    truck = [0, 0] # 배달 상자 ,수거 상자
    # 가장 먼 곳부터 이동하기
    for i in range(n-1, -1, -1):
        truck[0] += deliveries[i]
        truck[1] += pickups[i]
        # 상자를 모두 채울 때까지 이동
        while truck[0] > 0 or truck[1] > 0:
            # 상자가 남아있을 경우 cap 수만큼 내림
            truck [0] -= cap
            truck [1] -= cap
            answer += (i+1) * 2 # 다시 지점으로 돌아오므로 *2
    return answer
